cdist/exec/remote.py

```diff
@@ -58,27 +59,21 @@ def __init__(self,
                  target_host,
                  remote_exec,
                  remote_copy,
-                 base_path=None,
+                 base_path="/var/lib/cdist",
```

Comment:
While we are here already, can we change this to `/tmp/skonfig`? I don't really like that we expect that `/var/lib/` is always writeable. In ideal world we should use something like `<target-host>:$HOME/.skonfig/tmp/`, but I'm not aware good compatible way how to figure out remote user's `$HOME`. So using `/tmp/skonfig` feels like next best option, because (almost always) every user can write there.

Reply:
I was thinking about changing this path, too. I also find `/var/lib` special for primarily temporary data.
One issue with `/tmp` is that it may be mounted `noexec` and I feel like this may be a good idea in principle.
`/var` should always be writable IMO, since it's literally meant for *var*ying data.

Maybe `/var/tmp` would be a better compromise if we can assume that it is generally `rw,exec`…

Reply:
> /tmp is that it may be mounted noexec

If manifest, gencodes or explorers don't have execute bit set, they will be run with `/bin/sh -e`.

```
DEBUG: localhost: /tmp/tmpg9z867bg/421aa90e079fa326b6494f812ad13e79/data/conf/type/__ander_machine/manifest is executable, running it
TRACE: localhost: Local run: ['/home/ander/.cdist/type/__ander_machine/manifest']

```

versus without x bit:

```
DEBUG: localhost: /tmp/tmpxkui_fon/421aa90e079fa326b6494f812ad13e79/data/conf/type/__ander_machine/manifest is NOT executable, running it with /bin/sh -e
TRACE: localhost: Local run: ['/bin/sh', '-e', '/tmp/tmpxkui_fon/421aa90e079fa326b6494f812ad13e79/data/conf/type/__ander_machine/manifest']
```

But `/var/tmp/` maybe good compromise.

Reply:
+1 for splitting this off into a separate issue from my side.